spec/unit/pops/validator/validator_spec.rb

```diff
@@ -34,6 +38,61 @@ def validate(factory)
     expect(validate(parse('function ::aaa() {}'))).to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_NAME)
   end
 
+  it 'should raise error for illegal definition locations' do
+    expect(validate(parse('function aaa::ccc() {}', 'aaa/manifests/bbb.pp'))).to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('class bbb() {}', 'aaa/manifests/init.pp'))).to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('define aaa::bbb::ccc::eee() {}', 'aaa/manifests/bbb/ddd.pp'))).to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+  end
+
+  it 'should not raise error for legal definition locations' do
+    expect(validate(parse('function aaa::bbb() {}',      'aaa/manifests/bbb.pp'))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('define bbb() {}',      'aaa/manifests/site.pp'))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('class aaa() {}',      'aaa/manifests/init.pp'))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('class aaa() {}',      'manifests/site.pp'))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('function aaa::bbB::ccc() {}', 'aaa/manifests/bBb.pp'))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+    expect(validate(parse('function aaa::bbb::ccc() {}', 'aaa/manifests/bbb/CCC.pp'))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+  end
+
+  it 'should not raise error for class locations when not parsing a file' do
+    #nil file means eval or some other way to get puppet language source code into the catalog
+    expect(validate(parse('function aaa::ccc() {}', nil))).not_to have_issue(Puppet::Pops::Issues::ILLEGAL_DEFINITION_LOCATION)
+  end
+
+  context "with a manifest in our environment" do
```

Comment:
Picky - what is "our" environment? Do you mean the "effective" or "current" environment? (i.e. the one in use when compiling). 
Since the only thing being set at this level is an environment (`bar`) - maybe just say `In an environment`